psl/psl/tests/config/sources.rs

```diff
@@ -659,7 +717,86 @@ fn referential_integrity_default() {
 
     let config = parse_configuration(schema);
 
-    assert_eq!(config.referential_integrity(), Some(ReferentialIntegrity::ForeignKeys));
+    assert_eq!(config.relation_mode(), Some(RelationMode::ForeignKeys));
+}
+
+#[test]
+fn relation_mode_default() {
+    let schema = indoc! {r#"
+        datasource ps {
+          provider = "sqlserver"
+          url = "mysql://root:prisma@localhost:3306/mydb"
+        }
+
+        generator client {
+          provider = "prisma-client-js"
+          previewFeatures = ["relationMode"]
+        }
+    "#};
+
+    let config = parse_configuration(schema);
+
+    assert_eq!(config.relation_mode(), Some(RelationMode::ForeignKeys));
+}
+
+#[test]
+fn relation_mode_and_referential_integrity_can_coexist() {
+    let schema = indoc! {r#"
+        datasource ps {
+          provider = "sqlserver"
+          url = "mysql://root:prisma@localhost:3306/mydb"
+        }
+
+        generator client {
+          provider = "prisma-client-js"
+          previewFeatures = ["referentialIntegrity", "relationMode"]
+        }
+    "#};
+
+    let config = parse_configuration(schema);
+
+    assert_eq!(config.relation_mode(), Some(RelationMode::ForeignKeys));
+}
+
+#[test]
+fn relation_mode_prisma_has_precedence_over_referential_integrity() {
+    let schema = indoc! {r#"
+        datasource ps {
+          provider = "sqlserver"
+          url = "mysql://root:prisma@localhost:3306/mydb"
+          relationMode = "prisma"
+          referentialIntegrity = "foreignKeys"
+        }
+
+        generator client {
+          provider = "prisma-client-js"
+          previewFeatures = ["referentialIntegrity", "relationMode"]
+        }
+    "#};
+
+    let config = parse_configuration(schema);
+
+    assert_eq!(config.relation_mode(), Some(RelationMode::Prisma));
+}
```

Comment:
No need to change this in this PR, but that behavior does not feel obvious. If we have two conflicting settings specified for the same thing, my preference would be a validation error. But I think it's really an edge case and it's temporary.

Reply:
I agree - both things at the same time should not validate.
We will have this more and more over time, so we should have a good way to handle that.